ADAL/src/request/ADWebRequest+NSURLSession.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+@class ADWebRequest;
+@class ADWebResponse;
+
+typedef enum
+{
+    ADWebRequestGet,
+    ADWebRequestPost,
+} ADWebRequestMethodType;
+
+@interface ADWebRequest : NSObject
+{
+    NSURL * _requestURL;
+    NSMutableDictionary * _requestHeaders;
+    NSData * _requestData;
+    
+    NSHTTPURLResponse * _response;
+    NSMutableData * _responseData;
+    
+    NSUUID * _correlationId;
+    
+    NSUInteger _timeout;
+    
+    NSOperationQueue * _operationQueue;
+    
+    BOOL _isGetRequest;
+    
+    void (^_completionHandler)(NSError *, ADWebResponse *);
+}
+
+@property (strong, readonly, nonatomic)     NSURL               *URL;
+@property (copy, nonatomic)                 NSMutableDictionary *headers;
```

Comment:
Yeah, I think we should fix it in ADWebRequest as well. It’s not a huge deal because this library usually only accesses instance variables directly rather than through property accessors. Still, it’s wrong, and I think we should fix it.

On Jul 29, 2016, at 7:26 PM, Eric Tang <notifications@github.com<mailto:notifications@github.com>> wrote:


In ADAL/src/request/ADWebRequest+NSURLSession.h<https://github.com/msftberlin/azure-activedirectory-library-for-objc/pull/2#discussion_r72877551>:

> +    NSHTTPURLResponse * _response;
> +    NSMutableData * _responseData;
> +
> +    NSUUID * _correlationId;
> +
> +    NSUInteger _timeout;
> +
> +    NSOperationQueue * _operationQueue;
> +
> +    BOOL _isGetRequest;
> +
> +    void (^_completionHandler)(NSError *, ADWebResponse *);
> +}
> +
> +@property (strong, readonly, nonatomic)     NSURL               *URL;
> +@property (copy, nonatomic)                 NSMutableDictionary *headers;


The ADWebRequest+NSURLSession is a "copy" of ADWebRequest source code except the usage of NSURLSession, especially the declaration in the header files are pretty much the same. If this would be a problem, should we surface it back to main repository?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/msftberlin/azure-activedirectory-library-for-objc/pull/2/files/f8fff06aeb287b8237660a2228bfaaeb16ab8af0#r72877551>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAMWooo_HpttQOGFH7wyGBqUGJIeVcUAks5qapodgaJpZM4JP-vI>.